plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Core/Instance/Flat.hs

```diff
@@ -123,3 +123,43 @@ instance ( Flat ann
          ) => Flat (Program name uni fun ann) where
     encode (Program ann v t) = encode ann <> encode v <> encode t
     decode = Program <$> decode <*> decode <*> decode
+
+instance ( Closed uni
+         , uni `Everywhere` Flat
+         , Flat fun
+         ) => Flat (ETerm uni fun ) where
+    encode = \case
+        EVar      n    -> encodeTerm 0 <> encode n
+        EDelay    t    -> encodeTerm 1 <> encode t
+        ELamAbs   n t  -> encodeTerm 2 <> encode n <> encode t
```

Comment:
No Binder optimization here? :(

Reply:
Disregard what I said, we need to encode `n` since we are currently evaluating using uniques in cek environment

Reply:
Yes, this is not a debruijn ETerm!